addons/point_of_sale/models/pos_order.py

```diff
@@ -249,7 +249,7 @@ def _get_pos_anglo_saxon_price_unit(self, product, partner_id, quantity):
     is_total_cost_computed = fields.Boolean(compute='_compute_is_total_cost_computed',
         help="Allows to know if all the total cost of the order lines have already been computed")
     lines = fields.One2many('pos.order.line', 'order_id', string='Order Lines', states={'draft': [('readonly', False)]}, readonly=True, copy=True)
-    company_id = fields.Many2one('res.company', string='Company', required=True, readonly=True)
+    company_id = fields.Many2one('res.company', string='Company', required=True, readonly=True, index=True)
```

Comment:
@pivi-odoo Yeah for these two queries the index is not needed. `pos_reference` is more specific than `company_id` so in most cases if both fields are present in a `WHERE` clause the index on `pos_reference` will be hit. Except in the case of really weird conditions that I don't think can happen in standard, a bitmap scan using both is highly unlikely.

The thing is, the cardinality of the `pos.order` table can become pretty high in databases using PoS extensively. It's not rare to have something between 500K-1M orders. Also, those databases usually have multiple companies for shops in different places (there are some databases like that on the SaaS and on dedicated SH servers, mostly managed by FJA and the ps-tech). The cardinality is close to what we can see for `stock.move` and `account.move`, both having an index on `company_id` (`stock.move` since a long time ago, `account.move` since v16). And given that there are almost always ir.rules on `company_id`, we can find probably some queries that could hit the index.

That being said, it's probably true that on average, there is only one company for a given database with PoS, or the orders are unbalanced w.r.t the company, making the index mostly useless. Also, the default ordering of pos.order is `date_order desc, name desc, id desc` and there is an index on `date_order`. So a simple select query without sudo and explicit ordering will actually hit the `date_order` index backward instead of the `company_id` one. Stuff like loading up the ListView is not impacted by the addition of an index on `company_id` because of that.

All in all, I cannot really say that an index on `company_id` is useless in general. But I don't have data that can 
substantiate that and if we want to be conservative about index addition, then we can drop it.

Reply:
hmm yeah, I guess for the use case of the bigger database running multiple POS instances, each from their own company, an index on the `company_id` will be practical. Also will be beneficial in the fw-ports in the version where `company branches` were introduced.
Thanks for the explanation. :)